.github/workflows/lsif.yml

```diff
@@ -0,0 +1,23 @@
+name: Sourcegraph LSIF
+on:
+  push:
+  pull_request:
+jobs:
+  lsif:
+    runs-on: ubuntu-latest
+    name: "Upload LSIF"
+    steps:
+      - uses: actions/checkout@v2
+      - uses: coursier/cache-action@v6.3
+      - uses: coursier/setup-action@v1.1.2
+        with:
+          jvm: adopt:11
+          apps: lsif-java
+      - name: Generate LSIF
+        run: lsif-java index
+      - name: Upload LSIF data
+        uses: sourcegraph/lsif-upload-action@master
+        with:
+          endpoint: https://sourcegraph.com
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Does this need any special authorization, e.g. some OAuth app that needs to be registered, or does it work out-of-the-box?

Also are there potential security implications to using the GitHub token (which e.g. also permits write access) here? Maybe it might be a good idea to limit the permissions here to only include those needed.

Reply:
This value should work by making a new developer access token and adding it to the secrets of your repo/org. You shouldn't need to perform any additional flow after that.

We need repo scope of personal access tokens so that we can read the ones attached to your profile. [Here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/enterprise/cmd/frontend/internal/codeintel/httpapi/auth_github.go?L128) is the point in Sourcegraph server code that actually queries GitHub on your behalf if you'd like to see how it's actually used. I'm not sure if a lower scope would allow us the same permissions (and if so we'd definitely recommend using that instead).

In the future we may also expand our user permissions model so that authenticating directly with the code host will become unnecessary (though I don't have a timeline to give you at this point).

Reply:
Thanks for the note. GitHub Actions allows quite fine-grained scoping for the implicit `GITHUB_TOKEN` inside Actions (see [here](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token) and [here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions)), which I would prefer to a personal access token. What would be the minimal set of permissions needed here (ideally only some subset of read permissions)?

Reply:
According to these docs, I believe _read_ access on `repository-projects` would be sufficient.